### DISCUSSION TITLE
ngspice compatible Library of Transmission Line components

### DISCUSSION THREAD

**Discussion by tomhajjar** [ra3xdh/qucs_s#895] [General]
I've been working on some Transmission Line components that will work with ngspice. Most of the components use the Spice TLIne. I also found a lossless directional coupler that I made into a subcircuit, 

Attached is the project and the Library

![2024-08-19_115707](https://github.com/user-attachments/assets/1c7b7070-dbde-4097-ae03-64607cb96acc)
![2024-08-19_115634](https://github.com/user-attachments/assets/73e6046f-eb79-4bf4-a1df-76227e1aabe4)
![2024-08-19_115617](https://github.com/user-attachments/assets/7a9efe77-4826-4887-9b66-019e303dd50a)

[Transmission_Line_Lib_prj.zip](https://github.com/user-attachments/files/16662657/Transmission_Line_Lib_prj.zip)
[Transmission_Line_prj.zip](https://github.com/user-attachments/files/16662655/Transmission_Line_prj.zip)

**Comment by ra3xdh** [comment]
Thanks! I have planned the adding of this library in the next release. See #896

**Comment by tomhajjar** [comment]
I haven't figured out the usefulness of the "Transmission-Line Inverter" on page 521 of the ngspice manual. It models a coaxial transmission  line better than a single TLine alone does. I would add it to the Library if it was useful. The sub-circuit would need formulas for Z and Td.

![2024-08-20_130433](https://github.com/user-attachments/assets/7492dda1-c2c1-449a-b937-e4c3dfcc5290)

**Comment by tomhajjar** [comment]
@ra3xdh 

Can you send "Amp_MS.sch"? I have been unable to get the version I made using TLines to work like your example.


![2024-08-20_194237](https://github.com/user-attachments/assets/ec8d0ab5-c22a-42e8-80e1-a2bac84d19eb)

![amp-ms](https://github.com/user-attachments/assets/9da9a5f8-e3e5-4f85-9b3e-ba3c2f3721a9)

[BFP193W_amplifier_prj.zip](https://github.com/user-attachments/files/16684104/BFP193W_amplifier_prj.zip)

**Reply by ra3xdh** [reply]
Here is the BFP193W amplifier schematic with MS lines.

[BFP193_amp.zip](https://github.com/user-attachments/files/16728199/BFP193_amp.zip)
